plugins/chat/app/services/chat/action/publish_auto_removed_user.rb

```diff
@@ -8,29 +8,26 @@ module Action
     # this in staff actions so it's obvious why these users were
     # removed.
     class PublishAutoRemovedUser < Service::ActionBase
-      # @param [Symbol] event_type What caused the users to be removed,
+      # @param [Symbol] event What caused the users to be removed,
       #   each handler will define this, e.g. category_updated, user_removed_from_group
       # @param [Hash] users_removed_map A hash with channel_id as its keys and an
       #   array of user_ids who were removed from the channel.
-      option :event_type
+      option :event
       option :users_removed_map
 
       def call
         return if users_removed_map.empty?
 
-        users_removed_map.each do |channel_id, user_ids|
+        users_removed_map.each do |channel_id, all_user_ids|
           job_spacer = JobTimeSpacer.new
-          user_ids.in_groups_of(1000, false) do |user_id_batch|
-            job_spacer.enqueue(
-              Jobs::Chat::KickUsersFromChannel,
-              { channel_id: channel_id, user_ids: user_id_batch },
-            )
+          all_user_ids.in_groups_of(1000, false) do |user_ids|
+            job_spacer.enqueue(Jobs::Chat::KickUsersFromChannel, { channel_id:, user_ids: })
```

Comment:
This can actually be moved within the `if all_user_ids.any?` condition.

Reply:
Sure, I tried to keep the PR to only change what's necessary and tried my best not to change "style-related" code (even though I kinda did it here 😅).

Perf-wise, `in_groups_of` will be a no-op if `all_user_ids` is empty, so it doesn't change much.